hazelcast/include/hazelcast/client/sql/sql_result.h

```diff
@@ -100,6 +107,77 @@ class HAZELCAST_API sql_result : public std::enable_shared_from_this<sql_result>
         std::shared_ptr<sql_page> first_page_;
     };
 
+    /**
+     * Copy is allowed for convenience but it does shallow copy so it should be avoided.
+    */
+    class HAZELCAST_API page_iterator_sync
+    {
+    public:
+        using difference_type = void;
+        using value_type = std::shared_ptr<sql_page>;
+        using pointer = std::shared_ptr<sql_page>;
+        using reference = std::shared_ptr<sql_page>&;
+        using iterator_category = std::input_iterator_tag;
+
+        page_iterator_sync(page_iterator&&, std::chrono::milliseconds timeout);
```

Comment:
It might throw exception because of check_closed method, could you document it?

Reply:
Fixed.